src/internal/m365/collection/drive/delta_tree_test.go

```diff
@@ -1205,3 +1205,187 @@ func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_GenerateCollectables()
+				folderID("c1"): d.strPath(t, folderName(), folderName("c1")),
+			},
+			expect: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName()),
+				folderID("c1"): d.strPath(t, folderName(), folderName("c1")),
+			},
+		},
+		{
+			name: "updates cascade to unseen children",
+			collectables: map[string]collectable{
+				rootID: {currPath: d.fullPath(t)},
+				folderID(): {
+					prevPath: d.fullPath(t, folderName("old")),
+					currPath: d.fullPath(t, folderName()),
+				},
+			},
+			prevPaths: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("old")),
+				folderID("c1"): d.strPath(t, folderName("old"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("old"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("old"), folderName("c2"), folderName("c3")),
+			},
+			expect: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName()),
+				folderID("c1"): d.strPath(t, folderName(), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName(), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName(), folderName("c2"), folderName("c3")),
+			},
+		},
+		{
+			name: "updates cascade to unseen children - escaped path separator",
+			collectables: map[string]collectable{
+				rootID: {currPath: d.fullPath(t)},
+				folderID(): {
+					prevPath: d.fullPath(t, folderName("o/ld")),
+					currPath: d.fullPath(t, folderName("n/ew")),
+				},
+			},
+			prevPaths: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("o/ld")),
+				folderID("c1"): d.strPath(t, folderName("o/ld"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("o/ld"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("o/ld"), folderName("c2"), folderName("c3")),
+			},
+			expect: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("n/ew")),
+				folderID("c1"): d.strPath(t, folderName("n/ew"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("n/ew"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("n/ew"), folderName("c2"), folderName("c3")),
+			},
+		},
+		{
+			name: "tombstoned files get removed",
+			collectables: map[string]collectable{
+				rootID:     {currPath: d.fullPath(t)},
+				folderID(): {prevPath: d.fullPath(t, folderName("old"))},
+			},
+			prevPaths: map[string]string{
+				rootID:         d.strPath(t),
+				folderID():     d.strPath(t, folderName("old")),
+				folderID("c1"): d.strPath(t, folderName("old"), folderName("c1")),
+				folderID("c2"): d.strPath(t, folderName("old"), folderName("c2")),
+				folderID("c3"): d.strPath(t, folderName("old"), folderName("c2"), folderName("c3")),
+			},
+			expect: map[string]string{
+				rootID: d.strPath(t),
+			},
+		},
+		{
+			name: "mix of moved and tombstoned",
```

Comment:
could you please make a similar test but for multiple folder moves? For example:
```text
start with:
- root
  - old
    - c1
      - c2
        - c3
```
do the equivalent of:
```text
mv root/old root/folder
mkdir root/old
mkdir root/old/c1
mv root/folder/c1/c2 root/old/c1/c2
```

expect:
```text
- root
  - old
    - c1
      - c2
        - c3
  - folder
    - c1
```

Reply:
yeah, happy to!